src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -371,10 +370,10 @@ private bool TryRemoveInternal(TKey key, out TValue value, bool matchValue, TVal
                         continue;
                     }
 
-                    Node prev = null;
+                    Node? prev = null;
                     for (Node curr = tables._buckets[bucketNo]; curr != null; curr = curr._next)
                     {
-                        Debug.Assert((prev == null && curr == tables._buckets[bucketNo]) || prev._next == curr);
+                        Debug.Assert((prev == null && curr == tables._buckets[bucketNo]) || prev!._next == curr);
```

Comment:
this line looks like it could NRE (prev == null && curr != tables._buckets[bucketNo]) - but presumably that's an inconsistent state

Reply:
I see what you're saying, but it's fine.  This is really just saying "If I'm not on the first iteration of the loop, then validate that prev._next == curr".